### ISSUE TITLE
[BUG] Invalid memory access bug in `string_to_float_kernel`

### ISSUE THREAD

**Issue by ttnghia** [NVIDIA/spark-rapids-jni#1574]
I ran `compute_sanitizer` on `CAST_STRING` test and found this:
```
[ RUN      ] StringToFloatTests/0.Simple
========= Invalid __shared__ read of size 8 bytes
=========     at 0x1f70 in void spark_rapids_jni::detail::string_to_float_kernel<float, (int)256>(T1 *, unsigned int *, int *, int *, const char *, const int *, const unsigned int *, int)
=========     by thread (128,0,0) in block (1,0,0)
=========     Address 0x98 is out of bounds
```

**Comment by ttnghia** [comment]
CC @hyperbolic2346.

**Comment by hyperbolic2346** [comment]
The invalid access is here: https://github.com/NVIDIA/spark-rapids-jni/blob/512c9c9f2e83ed72f82ce269ec0dd2152a16ea5e/src/main/cpp/src/cast_string_to_float.cu#L420. `safe_count` is 19 and `_ipow` only has 19 entries. Investigating proper fix here.

**Comment by hyperbolic2346** [comment]
Fixed with #1591
